management/src/main/java/io/micronaut/management/health/monitor/HealthMonitorTask.java

```diff
@@ -85,24 +86,33 @@ void monitor() {
             .map(HealthIndicator::getResult)
             .collect(Collectors.toList());
 
-        Flux
-            .merge(resultPublishers)
+        Flux.merge(resultPublishers)
             .collectList()
-            .doOnError((e) -> {
+            .doOnError(e -> {
                 if (LOG.isErrorEnabled()) {
-                    LOG.error("Health monitor check failed with exception: " + e.getMessage(), e);
+                    LOG.error("Health monitor check failed with exception: {}", e.getMessage(), e);
                 }
                 currentHealthStatus.update(HealthStatus.DOWN.describe("Error occurred running health check: " + e.getMessage()));
             })
             .subscribe(healthResults -> {
-                java.util.Optional<HealthResult> firstDown = healthResults.stream().filter(r -> r.getStatus().equals(io.micronaut.health.HealthStatus.DOWN) || !r.getStatus().getOperational().orElse(true))
+                healthResults.forEach(healthResult -> {
```

Comment:
should not we execute this `forEach` only if ` if (LOG.isTraceEnabled() || LOG.isDebugEnabled()` evaluates to true